### PR TITLE
client, mod: Log meido vision into the mod-log

### LINKED ISSUE #774: Implement mod-log functionality for meidovision
Perhaps letting the userbase know when a meido is using meidovision in the mod-log may be a good idea?

### PR DESCRIPTION
**Pull request by Chiiruno**
Closes #774

### CODE REVIEW

File: client/posts/menu.ts [hunk 1]
```diff
@@ -232,11 +232,16 @@ function openMenu(e: Event) {
 
 // Fetch posts with the same IP on this board
 async function getSameIPPosts(m: Post): Promise<PostData[]> {
-	const res = await postJSON(`/api/same-IP/${m.id}`, null)
+	const res = await postJSON(`/api/same-IP`, {
+		lid: loginID(),
```
Comment (bakape) [inline client/posts/menu.ts]: Redundant. Login ID is transmitted through cookies. See `server::isLoggedIn()`.

File: go/src/meguca/db/admin.go [hunk 2]
```diff
@@ -11,7 +11,17 @@ import (
 )
 
 // Write moderation action to log
-func logModeration(tx *sql.Tx, e auth.ModLogEntry) error {
+func LogModeration(e auth.ModLogEntry) error {
```
Comment (bakape) [inline go/src/meguca/db/admin.go]: I'd prefer this not be exported out of the module. Please call logModeration from within the db module. Best to integrate this into `GetSameIPPosts()`.

File: go/src/meguca/db/admin.go [hunk 3]
```diff
@@ -11,7 +11,17 @@ import (
...
+	_, err := sq.Insert("mod_log").
+		Columns("type", "board", "id", "by", "length", "reason").
+		Values(int(e.Type), e.Board, e.ID, e.By, e.Length, e.Reason).
+		Exec()
+
+	return err
+}
+
+// Write moderation action to log atomically
+func logModerationA(tx *sql.Tx, e auth.ModLogEntry) error {
```
Comment (bakape) [inline go/src/meguca/db/admin.go]: Will also make this redundant.

File: go/src/meguca/server/admin.go [hunk 4]
```diff
@@ -40,6 +40,7 @@ var (
 	errBoardNameTaken   = common.ErrInvalidInput("board name taken")
 	errNoReason         = common.ErrInvalidInput("no reason provided")
 	errNoDuration       = common.ErrInvalidInput("no ban duration provided")
+	errMalformedRequest = common.ErrInvalidInput("malformed client request")
```
Comment (bakape) [inline go/src/meguca/server/admin.go]: Please provide a more specific error message. This will aid debugging in the future.

Comment (Chiiruno) [reply go/src/meguca/server/admin.go]: Any ideas on what kind of specific information this needs?

Comment (bakape) [reply go/src/meguca/server/admin.go]: Something descriptive. I'm not sure this message will be still needed after you rework the db side though. Best start there.

Comment (Chiiruno) [reply go/src/meguca/server/admin.go]: Yeah, I'm not using `errMalformedRequest` since I reverted to the old router line.

**Comment by Chiiruno** [comment]
@bakape How does this look?

File: go/src/meguca/db/admin.go [hunk 5]
```diff
@@ -372,47 +372,88 @@ func CanPerform(account, board string, action auth.ModerationLevel) (
...
+
+		// Add a mod-log entry detailing that a meido has used meido vision
+		var name string
+		pos, err := FindPosition(board, creds.UserID)
+
+		if err != nil {
+			return
+		}
+
+		switch pos {
```
Comment (bakape) [inline go/src/meguca/db/admin.go]: Why did you add this reason generation logic here? Is it really needed?

Comment (Chiiruno) [reply go/src/meguca/db/admin.go]: I can't say it's needed, but I thought it might be interesting to see if staff is looking at other staff.
As far as I can tell, this shouldn't leak any information and de-anonymize anything.

Comment (Chiiruno) [reply go/src/meguca/db/admin.go]: An alternative would be to do "requesting ${postID} history.
I don't really think that would have any particular negative effects either.

Comment (bakape) [reply go/src/meguca/db/admin.go]: The other staff thing seems pointless to me. The reasons is also unneeded. You can specify the post ID already with `auth.ModLogEntry.ID`. Thoughts?

Comment (Chiiruno) [reply go/src/meguca/db/admin.go]: I'll look into the post ID thing.
If you don't want the staff/reason stuff I'll remove it, if you don't care I'll keep it.
If you prefer the staff one or the postID one in the case of "don't care", please specify.

Comment (Chiiruno) [reply go/src/meguca/db/admin.go]: Oh, right. Okay, so you want the postID in the modlogentry.ID spot.
I don't want to deanonymize anons, so I'll leave out the staff stuff.

**Comment by Chiiruno** [comment]
@bakape Ready for merging again.

**Comment by Chiiruno** [comment]
I removed a now-pointless comment, rebased because it's too small to make another commit.

File: go/src/meguca/server/admin.go [hunk 6]
```diff
@@ -559,20 +559,31 @@ func assignStaff(w http.ResponseWriter, r *http.Request) {
...
 	}
+
 	board, _, ok := canModeratePost(w, r, id, auth.Janitor)
+
 	if !ok {
 		return
 	}
 
-	posts, err := db.GetSameIPPosts(id, board)
+	creds, ok := isLoggedIn(w, r)
```
Comment (bakape) [inline go/src/meguca/server/admin.go]: Needless. canModeratePost() returns user ID.

**Comment by Chiiruno** [comment]
ez
